fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/mixin/content/registry/MixinAbstractFurnaceBlockEntity.java

```diff
@@ -34,4 +36,19 @@
 	private static void fuelTimeMapHook(CallbackInfoReturnable<Map<Item, Integer>> info) {
 		FuelRegistryImpl.INSTANCE.apply(info.getReturnValue());
 	}
+
+	@Inject(at = @At("HEAD"), method = "canUseAsFuel", cancellable = true)
+	private static void canUseAsFuelHook(ItemStack stack, CallbackInfoReturnable<Boolean> info) {
+		info.setReturnValue(FuelRegistry.INSTANCE.get(stack.getItem()) != null);
+	}
+
+	@Inject(at = @At("HEAD"), method = "getFuelTime", cancellable = true)
+	protected void getFuelTimeHook(ItemStack fuel, CallbackInfoReturnable<Integer> info) {
```

Comment:
Can't all of these turn into 1 single mixin at AbstractFurnaceBlockEntity#createFuelTimeMap returning the cached map?

Reply:
It's a possibility. I'm not fond of it because it "leaks" an implementation detail (i.e. the cached map).

Reply:
We would have to return an immutable map in this case which may change the semantics of anyone using the furnace. As such I'd probably just add getters in the fuel registry